pkg/config/config_loader.go

```diff
@@ -31,3 +33,19 @@ func Load(target interface{}, loader SourcesProvider) error {
 	}
 	return yaml.Unmarshal(source, target)
 }
+
+// IkeProwLocalLoadableConfig holds absolute path to a local config file (located in ike-prow-plugins project) to be loaded
+type IkeProwLocalLoadableConfig struct {
+	AbsFilePath string
+}
+
+// Sources loads local config file that is located in ike-prow-plugins project structure
+func (i *IkeProwLocalLoadableConfig) Sources() []Source {
+	return []Source{func() ([]byte, error) {
+		file, err := ioutil.ReadFile(i.AbsFilePath)
```

Comment:
> I'm afraid this will not work when we run just the binary (or ship it as docker image). We have to make this static resource available somehow.

Good point - I didn't realize that...

> We have two options I can think of

there is a third option, but probably not the best - to get the file from GH.

> I would personally opt for the second option. I'm aware suggested library not really maintained anymore, but other OSIO services are using it without any issues so far. Fairly simple intro on how to set it up can be found here: http://rachbelaid.com/embedding-assets-in-go-project/

if it is used in other OSIO projects without any problem, then I'm fine with it

> I could think of creating a common place for static content for all the plugins. 

one common place was an option I was thinking about as well. 

> We can then also use it for templates. How about the following structure?

what kind of templates? Templates that are in the cluster directory or something else? I'm asking as I would keep them where they are - it makes more sense to me.

Reply:
Templates for messages